src/reactjs/useQueries.ts

```diff
@@ -118,7 +118,18 @@ export function useQueries<T extends any[]>({
   const mountedRef = React.useRef(false)
   const [, forceUpdate] = React.useState(0)
 
-  const queryClient = useQueryClient()
+  // Ensure that all queries have the same context
+  queries.forEach((curr, i) => {
+    const next = queries[i + 1]
+    if (next && curr.context !== next.context) {
+      throw new Error(
+        'useQueries requires all queries to have the same context'
+      )
+    }
+  })
+
+  const context = queries[0]?.context
+  const queryClient = useQueryClient({ context })
```

Comment:
since `useQueries` has the object syntax in v4, let's please do:

```
useQueries({
  queries: [...],
  context: context
})
```

on type level, we could `Omit` the `context` property from the object we pass to each `query`

Reply:
Sorry for the delay. This is completed and pushed up.